enterprise/internal/licensing/features.go

```
@@ -126,6 +126,43 @@ func (f *FeaturePrivateRepositories) Check(info *Info) error {
 	return NewFeatureNotActivatedError(fmt.Sprintf("The feature %q is not activated in your Sourcegraph license. Upgrade your Sourcegraph subscription to use this feature.", f.FeatureName()))
 }
 
+type FeatureLLMProxyTier string
+
+const (
+	FeatureLLMProxyTierDefault FeatureLLMProxyTier = "default"
+)
+
+// FeatureLLMProxy is what degree of LLM-proxy access is enabled for this license.
+type FeatureLLMProxy struct {
+	// TODO: Figure out specific tiers we will have.
+	Tier FeatureLLMProxyTier
+}
+
+func (*FeatureLLMProxy) FeatureName() string { return "llm-proxy" }
+
+func (f *FeatureLLMProxy) Check(info *Info) error {
+	if info == nil {
+		return NewFeatureNotActivatedError(fmt.Sprintf("The feature %q is not activated because it requires a valid Sourcegraph license. Purchase a Sourcegraph subscription to activate this feature.", f.FeatureName()))
+	}
```

Should we return a similar error if service accounts feature flag is not turned on?